tests/testthat/test-dplyr-joins.R

```diff
@@ -685,8 +716,48 @@ test_that("RIGHT JOIN - NA matches", {
     expect_contains("warn")
 })
 
+test_that("RIGHT JOIN - unmatched error", {
+
+ right_join(x            = x4,
+            y            = y4,
+            relationship = "many-to-many",
+            by           = "id2",
+            unmatched    = "error") |>
+    expect_no_error()
+
+ right_join(x            = x4,
+            y            = y4,
+            relationship = "many-to-many",
+            by           = "id1=id2",
+            unmatched    = "error") |>
+    expect_no_error()
+
+ right_join(x            = x1,
+            y            = y1,
+            relationship = "many-to-one",
+            by           = "id",
+            unmatched    = "error") |>
+    expect_error()
+
+ right_join(x            = x4,
+           y            = y4,
+           relationship = "many-to-many",
+           by           = c("id2", "x"),
+           unmatched    = "error") |>
+   expect_error()
+
+ right_join(x            = x4,
+           y            = y4,
+           relationship = "many-to-many",
+           by           = c("id1=id2", "id2=id"),
+           unmatched    = "error") |>
+   expect_no_error()
```

Comment:
This should not throw an error, but rather store a warning message (due to the duplicate names issue we discussed -in this case id2 is already in y so that id cannot be renamed to id2)